# Análise de Sentimentos em texto 🧠😊🤖💕

### Crie um programa que realiza a análise de sentimento de um texto fornecido pelo usuário. O programa deve utilizar uma biblioteca de processamento de linguagem natural (NLP) para identificar se o sentimento do texto é positivo, negativo ou neutro.

In [1]:
from textblob import TextBlob
from deep_translator import GoogleTranslator

In [3]:
# O textblob funciona melhor em inglês, por isso é recomendado traduzir de pt para en

def translate_to_english(text):
    translator = GoogleTranslator(source="pt", target="en")
    translation = translator.translate(text)
    return translation

def analyze_sentiment(text):
    text_in_en = translate_to_english(text)
    blob = TextBlob(text_in_en)
    sentiment = blob.sentiment
    return sentiment

In [4]:
texto = "Comecei a gostar de futebol apenas com 12 anos, gostar ao ponto de responsabilizar jogos pelos meus estados emocionais. Cresci falando que torcia para o Vasco da Gama e que o amava, isso continua até hoje, mesmo nunca tendo ido a São Januário. Espero um dia poder ir e gritar aos 4 cantos do mundo que a torcida do Vascão está sempre tão linda."
sentimento_detectado = analyze_sentiment(texto)
print(f'Texto: {texto}\nSentimento: {sentimento_detectado}')

Texto: Comecei a gostar de futebol apenas com 12 anos, gostar ao ponto de responsabilizar jogos pelos meus estados emocionais. Cresci falando que torcia para o Vasco da Gama e que o amava, isso continua até hoje, mesmo nunca tendo ido a São Januário. Espero um dia poder ir e gritar aos 4 cantos do mundo que a torcida do Vascão está sempre tão linda.
Sentimento: Sentiment(polarity=0.39999999999999997, subjectivity=0.7416666666666667)


### Polaridade:
Definição: A polaridade indica o quão positivo ou negativo é o sentimento expresso em um texto.
Intervalo: Varia de -1 (muito negativo) a 1 (muito positivo).

### Subjetividade:
Definição: A subjetividade mede o quanto o texto é subjetivo ou objetivo.
Intervalo: Varia de 0 (muito objetivo) a 1 (muito subjetivo).



---

No meu resultado anterior, Sentiment(polarity=0.4, subjectivity=0.74) indica que o texto é ligeiramente positivo e bastante subjetivo.